iOS: Update synccontext errors and completions

phvannor opened the pull request:
Fix completion blocks not occurring on the specified callback queue.

Surface operation write errors to callers when they occur post item update.

brettsam commented on sdk/iOS/src/MSSyncContext.m:
```diff
@@ -126,7 +125,9 @@ -(void) syncTable:(NSString *)table
-            completion(nil, error);
+            [self.callbackQueue addOperationWithBlock:^{
+                completion(nil, error);
+            }];
```
Can our unit tests verify these are all coming back on the correct queue?

brettsam commented:
Other than my unit test comment, everything else looks good.

phvannor commented:
Added check of queue into insert tests, and updated push callback mechanism to match.  Since I'm updating the rest (see my other PR), I'll do the same for update/delete/etc in future ones.

brettsam commented:
Looks good.
